### PR TITLE
{Test} Make tests more robust

### PR DESCRIPTION
**Pull request by fengzhou-msft**
**Description**  

1. Add double quotes for file path parameters to make it still work when there're spaces in the file path.
2. Use `os.path.join()` to join file path, so the tests work across Windows and Posix systems. 
3. Add `--admin-username` in `vm/vmss create` explicitly due to #14265 

**Testing Guide**  

**History Notes**  

[Component Name 1] BREAKING CHANGE: az command a: Make some customer-facing breaking change.  
[Component Name 2] az command b: Add some customer-facing feature.

---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [ ] The PR title and description has followed the guideline in [Submitting Pull Requests](https://github.com/Azure/azure-cli/tree/dev/doc/authoring_command_modules#submitting-pull-requests).

- [ ] I adhere to the [Command Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md).

### CODE REVIEW

**Comment by yonzhan** [comment]
Test

**Comment by jiasli** [comment]
Nice catch!

File: src/azure-cli/azure/cli/command_modules/batch/tests/latest/test_batch_data_plane_commands.py [hunk 1]
```diff
@@ -320,7 +320,7 @@ def test_batch_pools_and_nodes(
 
         # test create pool from JSON file
         self.kwargs['json'] = self._get_test_data_file('batch-pool-create.json').replace('\\', '\\\\')
-        self.batch_cmd('batch pool create --json-file {json}')
+        self.batch_cmd('batch pool create --json-file "{json}"')
```
Comment (fengzhou-msft) [inline src/azure-cli/azure/cli/command_modules/batch/tests/latest/test_batch_data_plane_commands.py:323]: @bgklein can you review changes in `test_batch_data_plane_commands.py` and approve if it looks good to you?

**Comment by zhoxing-ms** [comment]
@fengzhou-msft May I ask why not re-record the yaml files? I think it's better to run it again in live mode to make sure it doesn't affect the test results.

**Comment by fengzhou-msft** [comment]
> @fengzhou-msft May I ask why not re-record the yaml files? I think it's better to run it again in live mode to make sure it doesn't affect the test results.

The purpose is to make tests pass in different environments such as on Windows or in Docker and considering cases when the source code is under a directory that contains spaces. Re-recording does not help in this case.

**Pull request merged**